File: src/lib/tauri-api/workspace-scripts.ts
import type { TerminalSession } from '../../types/terminal';
import type { ForgeWorkspaceConfig } from '../../types/workspace-scripts';
import { invokeCommand } from './client';

export function getWorkspaceForgeConfig(workspaceId: string): Promise<ForgeWorkspaceConfig> {
  return invokeCommand<ForgeWorkspaceConfig>('get_workspace_forge_config', { workspaceId });
}

export function runWorkspaceSetup(workspaceId: string): Promise<TerminalSession[]> {
  return invokeCommand<TerminalSession[]>('run_workspace_setup', { workspaceId });
}

export function startWorkspaceRunCommand(workspaceId: string, commandIndex: number): Promise<TerminalSession> {
  return invokeCommand<TerminalSession>('start_workspace_run_command', { workspaceId, commandIndex });
}

export function restartWorkspaceRunCommand(workspaceId: string, commandIndex: number): Promise<TerminalSession> {
  return invokeCommand<TerminalSession>('restart_workspace_run_command', { workspaceId, commandIndex });
}

export function stopWorkspaceRunCommands(workspaceId: string): Promise<TerminalSession[]> {
  return invokeCommand<TerminalSession[]>('stop_workspace_run_commands', { workspaceId });
}

File: src-tauri/src/commands/workspace_scripts.rs
use tauri::State;

use crate::models::{ForgeWorkspaceConfig, TerminalSession, WorkspaceHookInspector};
use crate::services::{hook_service, workspace_script_service};
use crate::state::AppState;

#[tauri::command]
pub fn get_workspace_forge_config(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<ForgeWorkspaceConfig, String> {
    workspace_script_service::get_workspace_forge_config(&state, &workspace_id)
}

#[tauri::command]
pub fn run_workspace_setup(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<Vec<TerminalSession>, String> {
    workspace_script_service::run_workspace_setup(&state, &workspace_id)
}

#[tauri::command]
pub fn start_workspace_run_command(
    state: State<'_, AppState>,
    workspace_id: String,
    command_index: usize,
) -> Result<TerminalSession, String> {
    workspace_script_service::start_workspace_run_command(&state, &workspace_id, command_index)
}

#[tauri::command]
pub fn restart_workspace_run_command(
    state: State<'_, AppState>,
    workspace_id: String,
    command_index: usize,
) -> Result<TerminalSession, String> {
    workspace_script_service::restart_workspace_run_command(&state, &workspace_id, command_index)
}

#[tauri::command]
pub fn stop_workspace_run_commands(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<Vec<TerminalSession>, String> {
    workspace_script_service::stop_workspace_run_commands(&state, &workspace_id)
}

#[tauri::command]
pub fn get_workspace_hook_inspector(
    state: State<'_, AppState>,
    workspace_id: String,
) -> Result<WorkspaceHookInspector, String> {
    hook_service::get_workspace_hook_inspector(&state, &workspace_id)
}
